src/test/java/de/zalando/aruha/nakadi/controller/EventTypeControllerTest.java

```diff
@@ -264,9 +266,23 @@ public void whenDeleteEventTypeAndTopicDeletionExceptionThen503() throws Excepti
     }
 
     @Test
+    public void whenCreateEventTypeWithWrongPartitionKeyFieldsThen422() throws Exception {
+
+        final EventType eventType = buildEventType(randomValidEventTypeName(), new JSONObject("{ \"price\": 1000 }"),
+                Collections.singletonList("blabla"));
+
+        Mockito.doReturn(eventType).when(eventTypeRepository).findByName(eventType.getName());
+        Mockito.doThrow(new TopicDeletionException("dummy message", null)).when(topicRepository).deleteTopic(
```

Comment:
I could not understand why is this mock `doThrow` needed. Are you sure it's needed?